tests/org.jboss.tools.runtime.as.ui.bot.test/src/org/jboss/tools/runtime/as/ui/bot/test/parametized/server/ServerRuntimesTest.java

```diff
@@ -0,0 +1,181 @@
+import java.util.List;
+
+import org.jboss.reddeer.eclipse.jdt.ui.preferences.JREsPreferencePage;
+import org.jboss.reddeer.junit.internal.runner.ParameterizedRequirementsRunnerFactory;
+import org.jboss.reddeer.junit.runner.RedDeerSuite;
+import org.jboss.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
+import org.jboss.tools.cdk.reddeer.requirements.DisableSecureStorageRequirement.DisableSecureStorage;
+import org.jboss.tools.runtime.as.ui.bot.test.Activator;
+import org.jboss.tools.runtime.as.ui.bot.test.SuiteConstants;
+import org.jboss.tools.runtime.as.ui.bot.test.download.RuntimeDownloadTestUtility;
+import org.jboss.tools.runtime.as.ui.bot.test.reddeer.Runtime;
+import org.jboss.tools.runtime.as.ui.bot.test.reddeer.util.DetectRuntimeTemplate;
+import org.jboss.tools.runtime.as.ui.bot.test.reddeer.util.OperateServerTemplate;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.Parameterized.UseParametersRunnerFactory;
+
+
+
+/**
+ * This test is trying to optimize and clean up the huge number of tests that were here. 
+ * 
+ * Pre-reqs:
+ *   1) -Dreq.java6.home={java6home}
+ *   2) -Dreq.java7.home={java7home}
+ *   3) -Dreq.java8.home={java8home}
+ *   4) -Djboss.org.username={yourusername}
+ *   5) -Djboss.org.password={hunter2}
+ *   6) -Druntimes.suite.scope={smoke | latestMajors | allFree | all}
+ *   
+ *  Items 4 and 5 are optional and only necessary if testing runtimes that 
+ *  require $0 subscription. 
+ *  
+ *  This test no longer needs to download any runtimes in advance. It will download them all via UI.
+ *  
+ *  There are 3 primary tests here:
+ *    1) acquireAndDetect	(download the runtime via ui and make sure it appears)
+ *    2) detect				(use the just-downloaded fs-location and use runtime detection)
+ *    3) operate			(import via rt-detection and start/stop it)
+ *    
+ *  JRE requirement is no longer necessary. We simply make sure to add a JRE
+ *  for java 6, 7, and 8 in advance, and check to make sure that the given server
+ *  starts without the user needing to customize or change the jre at all. 
+ *  
+ *  Adding new runtimes to the test is as easy as modifying 
+ *  ServerRuntimeUIConstants. You should:
+ *    1) Declare a constant representing the UI string to download a given runtime
+ *    2) Add that constant to the various arrays it should belong to (free, smoke, etc)
+ *    3) Add a line to the initialize() method representing the various expected dl-rt values
+ */
+
+
+@RunWith(RedDeerSuite.class)
+@UseParametersRunnerFactory(ParameterizedRequirementsRunnerFactory.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)//first acquireAndDetect, then detect, then operate
+@DisableSecureStorage
+public class ServerRuntimesTest {
+
+    @Parameters
```

Comment:
add `(name="{0}") so we can see in junit results for which runtime the test failed?